test/e2e/scenario/rum/errors.scenario.ts

```diff
@@ -121,35 +121,82 @@ describe('rum errors', () => {
         expect(browserLogs.length).toEqual(0)
       })
     })
+
+  // Ignore this test on Safari and firefox untill we upgrade because:
+  // - Safari < 15 don't report the property disposition
+  // - Firefox < 99 don't report csp violation at all
+  // TODO: Remove this condition when upgrading to Safari 15 and Firefox 99 (see: https://datadoghq.atlassian.net/browse/RUM-1063)
+  if (!((getBrowserName() === 'safari' && getPlatformName() === 'macos') || getBrowserName() === 'firefox')) {
+    createTest('send CSP violation errors')
+      .withRum()
+      .withBody(
+        createBody(`
+      const script = document.createElement('script');
+      script.src = "https://example.com/foo.js"
+      document.body.appendChild(script)
+      `)
+      )
+      .run(async ({ intakeRegistry, baseUrl }) => {
+        const button = await $('button')
+        await button.click()
+
+        await flushEvents()
+
+        expect(intakeRegistry.rumErrorEvents.length).toBe(1)
+        expectError(intakeRegistry.rumErrorEvents[0].error, {
+          message: /^csp_violation: 'https:\/\/example\.com\/foo\.js' blocked by 'script-src(-elem)?' directive$/,
+          source: 'report',
+          stack: [
+            /^script-src(-elem)?: 'https:\/\/example\.com\/foo\.js' blocked by 'script-src(-elem)?' directive of the policy/,
+            `  at <anonymous> @ ${baseUrl}/:`,
+          ],
+          handling: 'unhandled',
+          csp: {
+            disposition: 'enforce',
+          },
+        })
+        await withBrowserLogs((browserLogs) => {
+          expect(browserLogs.length).toEqual(1)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏what about adding an expect on the log message or a comment to make more explicit which log we are expecting?

Reply:
hmm, I'm not actually sure there is value to check that the browser logs in this case.

I wrote this test looking at the other ones on this file and I think this check is meant to verify that we do not generate any other errors unintentionally.

Also, error message tend to be widely inconsistent between browser vendor, which will make this test even more complex to make work on Safari and FireFox

Reply:
> I wrote this test looking at the other ones on this file and I think this check is meant to verify that we do not generate any other errors unintentionally.

Oh I see, IIRC we do those assertions to avoid to have these expected errors logged in the test output (see [test teardown](https://github.com/DataDog/browser-sdk/blob/main/test/e2e/lib/framework/createTest.ts#L217)).

So yes, no need to add more details about the log is generated by the test.